### PR TITLE
Display the current version in the settings sidebar. Closes #1061

### LINKED ISSUE #1061: Version of existing instance
Is there a way to find out the version number of the currently running Cachet instance?

### PR DESCRIPTION
**Pull request by jbrooksuk**
We don't really have anywhere else to put the version for now, so this will do, plus it's out of the way.

image

I've chosen to create a `VERSION` file in the root because this makes it easier for people to open the repo and immediately see the value. The defined version is a constant because this will make it easier to check versions in the future.

Ping @cachethq/owners

### CODE REVIEW

**Comment by TheoBearman** [comment]
Personally, I would move this to the bottom of the dashboard navigation bar.

**Comment by jbrooksuk** [comment]
> Personally, I would move this to the bottom of the dashboard navigation bar.

I thought about that, but we always have issues with shallow screens. It'd probably move there once we have a redesigned dashboard.

**Comment by TheoBearman** [comment]
Then what about the footer on the status page? I just think it looks too out of place on the settings page.

**Comment by jbrooksuk** [comment]
> Then what about the footer on the status page?

Nope, I wouldn't want to publicly expose the version of Cachet like that.

**Comment by TheoBearman** [comment]
In that case, I guess where you have put it is the best place then.

**Comment by jbrooksuk** [comment]
> In that case, I guess where you have put it is the best place then.

Well, it's the best we can do for now.

File: VERSION [hunk 1]
```diff
@@ -0,0 +1 @@
+2.0.0-RC1
```
Comment (GrahamCampbell) [inline VERSION]: this is wrong

Comment (GrahamCampbell) [inline VERSION]: should be `2.0.0-dev`

Comment (GrahamCampbell) [inline VERSION]: we need to update it when we actually do a tag, then again straight after to `-dev`

File: app/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -28,6 +28,8 @@ public function boot(Dispatcher $dispatcher)
             return Dispatcher::simpleMapping($command, 'CachetHQ\Cachet', 'CachetHQ\Cachet\Handlers');
         });
 
+        define('CACHET_VERSION', trim(file_get_contents($this->app->basePath().'/VERSION')));
```
Comment (GrahamCampbell) [inline app/Providers/AppServiceProvider.php]: this will fail in some cases

Comment (GrahamCampbell) [inline app/Providers/AppServiceProvider.php]: for example, if we run the tests more than once

Comment (GrahamCampbell) [inline app/Providers/AppServiceProvider.php]: oh wow, yeh, already failing

Comment (GrahamCampbell) [inline app/Providers/AppServiceProvider.php]: best option here is to set this constant in the early bootstrap process

Comment (GrahamCampbell) [inline app/Providers/AppServiceProvider.php]: https://github.com/cachethq/Cachet/blob/master/bootstrap/autoload.php

Comment (GrahamCampbell) [inline app/Providers/AppServiceProvider.php]: move this code into there

**Comment by jbrooksuk** [comment]
Thanks for finishing this @GrahamCampbell :)

**Comment by GrahamCampbell** [comment]
No problem. :)

**Comment by joecohens** [comment]
:heart_eyes:

**Comment by jbrooksuk** [comment]
We just need to remember to update this file now.

**Pull request merged**